### PR TITLE
account pkg: To remove errwrap, add unit testing and enhance error messages

### PR DESCRIPTION
**Pull request by stack72**
Part of: #10 
Part of: #76 

- [x] Account Package
  - [x] Unit tests
  - [x] Error Messages

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -40,3 +40,7 @@
 [[constraint]]
   branch = "master"
   name = "golang.org/x/crypto"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/pkg/errors"
```
Comment (sean-) [inline Gopkg.toml]: Missing newline

File: Gopkg.lock [hunk 2]
```diff
@@ -7,17 +7,23 @@
   packages = ["."]
   revision = "d5467c17e7afe8d8f08f556c6c811a50c3feb28d"
 
+[[projects]]
+  name = "github.com/davecgh/go-spew"
+  packages = ["spew"]
+  revision = "346938d642f2ec3594ed81d874461961cd0faa76"
+  version = "v1.1.0"
```
Comment (sean-) [inline Gopkg.lock]: Why is `spew` being used?  I'm guessing there's some uncommitted code locally that `dep` picked up.

File: account/config.go [hunk 3]
```diff
@@ -55,7 +55,7 @@ type UpdateConfigInput struct {
 func (c *ConfigClient) Update(ctx context.Context, input *UpdateConfigInput) (*Config, error) {
 	path := fmt.Sprintf("/%s/config", c.client.AccountName)
 	reqInputs := client.RequestInput{
-		Method: http.MethodPut,
+		Method: http.MethodPost,
```
Comment (sean-) [inline account/config.go]: 1) Why the change from `PUT` to `POST`?

2) Someday we should come through and replace the `fmt.Sprintf()` calls with `path.Join("/", c.client.AccountName, "config"`.  Not now, but at some point...

File: Gopkg.lock (hunk 2 above)
Comment (stack72) [reply Gopkg.lock]: I believe this is actually part of Errwrap dependencies - when we remove errwrap, it will go away

Comment (stack72) [reply Gopkg.lock]: Oh maybe not....

File: account/config.go (hunk 3 above)
Comment (stack72) [reply account/config.go]: So the change from PUT to POST was to follow the convention of the other Update API endpoints in this package which are all POST

Comment (stack72) [reply account/config.go]: I reverted this - I am not sure this is the correct thing to do - I shouldn't change an endpoint that is already in place :)
